src/EditorFeatures/CSharpTest/GoToNextAndPreviousMember/CSharpGoToNextAndPreviousMemberTests.cs

```diff
@@ -0,0 +1,419 @@
+        public void StopsAtIndexerWithAccessors()
+        {
+            var code = @"
+class C
+{
+    $$void M1() { }
+
+    [||]int this[int i]
+    {
+        get { return 42; }
+        set { }
+    }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void SkipsIndexerAccessors()
+        {
+            var code = @"
+class C
+{
+    void M1() { }
+
+    $$int this[int i]
+    {
+        get { return 42; }
+        set { }
+    }
+
+    [||]void M2() { }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void StopsAtEventWithAddRemove()
+        {
+            var code = @"
+class C
+{
+    $$void M1() { }
+
+    [||]event EventHandler E
+    {
+        add { }
+        remove { }
+    }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void SkipsEventAddRemove()
+        {
+            var code = @"
+class C
+{
+    void M1() { }
+
+    $$event EventHandler E
+    {
+        add { }
+        remove { }
+    }
+
+    [||]void M2() { }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void FromInsideMethod()
+        {
+            var code = @"
+class C
+{
+    void M1()
+    {
+        $$System.Console.WriteLine();
+    }
+
+    [||]void M2() { }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void NextFromBetweenMethods()
+        {
+            var code = @"
+class C
+{
+    void M1() { }
+
+    $$
+
+    [||]void M2() { }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void PreviousFromBetweenMethods()
+        {
+            var code = @"
+class C
+{
+    [||]void M1() { }
+
+    $$
+
+    void M2() { }
+}";
+
+            AssertNavigated(code, next: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void NextFromBetweenMethodsInTrailingTrivia()
+        {
+            var code = @"
+class C
+{
+    void M1()
+    {
+    } $$
+
+    [||]void M2() { }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        public void PreviousFromBetweenMethodsInTrailingTrivia()
+        {
+            var code = @"
+class C
+{
+    [||]void M1()
+    {
+    } $$
+
+    void M2() { }
+}";
+
+            AssertNavigated(code, next: false);
```

Comment:
Maybe add a test for a non-empty file where the caret is outside all types and namespaces?